api/src/test/java/org/openmrs/validator/VisitValidatorTest.java

```diff
@@ -448,4 +455,78 @@ public void validate_shouldFailValidationIfFieldLengthsAreNotCorrect() {
 		new VisitValidator().validate(visit, errors);
 		assertEquals(true, errors.hasFieldErrors("voidReason"));
 	}
+	
+	@Test
+	public void validate_shouldFailValidationIfVisitDateBeforeBirthDate() {
+		Visit visit = new Visit();
+		Patient patient = new Patient();
+		calendar.set(1974, 4, 8);
+		patient.setBirthdate(calendar.getTime());
+		patient.setBirthdateEstimated(false);
+		visit.setPatient(patient);
+		calendar.set(1974, 4, 7);
+		visit.setStartDatetime(calendar.getTime());
+		Errors errors = new BindException(visit, "visit");
+		
+		new VisitValidator().validate(visit, errors);
+		
+		assertThat(errors, hasFieldErrors("startDatetime", "Visit.startDateCannotFallBeforeTheBirthDateOfTheSamePatient"));
+	}
+	
+	@Test
+	public void validate_shouldAcceptValidationIfVisitDateWithinEstimatedBirthDateMarginOfError() {
```

Comment:
use `Pass` instead of `Accept` for the positive test cases as this is what we have in other tests, rename to

`validate_shouldPassValidationIfVisitDateOnEstimatedBirthDatesGracePeriod`

to be close to naming of above test cases.